src/libraries/System.Private.CoreLib/src/System/ArraySegment.cs

```diff
@@ -276,6 +279,29 @@ private void ThrowInvalidOperationIfDefault()
             }
         }
 
+        private string DebuggerToString()
+        {
+            if (typeof(T) == typeof(char))
+            {
+                return new string(new ReadOnlySpan<char>(ref Unsafe.As<T, char>(ref _array![_offset]), _count));
+            }
+
+            return $"System.ArraySegment<{typeof(T).Name}>[{_count}]";
+        }
+
+        private sealed class ArraySegmentDebugView<U>
+        {
+            private readonly U[] _array;
+
+            public ArraySegmentDebugView(ArraySegment<U> arraySegments)
+            {
+                _array = arraySegments.ToArray();
+            }
+
+            [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+            public U[] Items => _array;
+        }
```

Comment:
```suggestion
        private sealed class ArraySegmentDebugView(ArraySegment<T> array)
        {
            [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
            public T[] Items => array.AsSpan().ToArray();
        }
```

Reply:
Have we started to roll-in primary constructor in runtime?

Reply:
Yes, for simple uses like debugger type proxies.